core/cancellation_exception.h

```diff
@@ -0,0 +1,31 @@
+ /** cancellation_exception.h                                               -*- C++ -*-
+    Guy Dumais, 5 October 2016
+    Copyright (c) 2016 Datacratic Inc.  All rights.
+
+    This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+    
+    Exception class to use to return procedure cancellation.
+*/
+
+#pragma once
+
+#include "mldb/arch/exception.h"
+#include "mldb/types/any.h"
+#include "mldb/types/string.h"
+
+namespace MLDB {
+
+struct CancellationException: public Exception {
```

Comment:
This should derive from SilentException, its methods should not be inline (there should be a .cc file), the details should be handled opaquely with a unique_ptr<Itl> and we should not need to include the any or string headers.  All this to make sure that the ABI is as tidy as possible, as this will be an API header.